action/protocol/staking/handler_stake_migrate.go

```diff
@@ -75,12 +73,16 @@ func (p *Protocol) handleStakeMigrate(ctx context.Context, act *action.MigrateSt
 	nonceUpdated = true
 	if excReceipt.Status != uint64(iotextypes.ReceiptStatus_Success) {
 		revertSM()
-		return nil, nil, 0, nonceUpdated, errors.Errorf("execution failed with status %d", excReceipt.Status)
+		nonceUpdated = false
+		return excReceipt.Logs(), excReceipt.TransactionLogs(), excReceipt.GasConsumed + insGas, nonceUpdated, &handleError{
+			err:           errors.Errorf("staking contract failure: %s", excReceipt.ExecutionRevertMsg()),
+			failureStatus: iotextypes.ReceiptStatus(excReceipt.Status),
+		}
 	}
 	// add sub-receipts logs
 	actLogs = append(actLogs, excReceipt.Logs()...)
 	transferLogs = append(transferLogs, excReceipt.TransactionLogs()...)
-	return actLogs, transferLogs, excReceipt.GasConsumed + insGas, nonceUpdated, nil
+	return actLogs, transferLogs, insGas, nonceUpdated, nil
```

Comment:
`return actLogs, transferLogs, insGas, true, nil`
it seems that if `err != nil`, `nonceUpdated` is always `false`, else it is always true.

Reply:
yeah, seems like nonceUpdated is not necessary, and just not update nonce again if returned error == nil